src/commonMain/kotlin/fr/acinq/lightning/serialization/v4/Serialization.kt

```diff
@@ -305,6 +305,7 @@ object Serialization {
         writeDelimited(pubkeyScript.toByteArray())
         writeNumber(localAmount.toLong())
         writeNumber(remoteAmount.toLong())
+        writeNumber(htlcAmount.toLong())
```

Comment:
Sane here, you must change the discriminator line 303 to `write(0x02)` since you introduce a new field.

Reply:
Also fixed in [99ff0f6](https://github.com/ACINQ/lightning-kmp/pull/568/commits/99ff0f693fc569bb89008a97791f5340d6b8478d).